src/Utils/Rotation.cpp

```diff
@@ -312,15 +312,335 @@ void utils::Rotation::checkUnitary()
 #ifndef SKIP_ASSERT
     double sqrtNorm = static_cast<double>(this->squaredNorm());
     bool checkUnit(fabs(sqrtNorm - 3.) < 1e-4);
-    if (!checkUnit){
-        // Make calulation here
-        utils::Matrix3d mat;
+    if (!checkUnit)
+    {
+
+        // Code addapted from https://github.com/brainexcerpts/3x3_polar_decomposition/blob/master/src/polar_decomposition_3x3.inl
+
+        // Placer inf_norm and one norm in a function
+        // créer fonction polar_decomposition et déplacer dans Matrix3d
+
+        float tolerance = 1e-6;
+        utils::Matrix3d M = this->block(0, 0, 3, 3);
+        utils::Matrix3d R;
+        utils::Matrix3d S;
+        utils::Matrix3d Mk;
+        utils::Matrix3d Ek;
+        float det, E_one_norm;
+        bool use_svd = false;
+
+        
+        // Mk = mat^T
+        Mk = M.transpose();
+        
+        //M one-norm
+        float M_one_norm = 0.0f;
+        for (int i = 0; i < 3; i++)
```

Comment:
For sake of consistency, use an unsigned int instead of an int. The unsigned int used in biorbd is `size_t`, therefore the for loop should read:
`for (size_t i = 0; i<3; ++i)` 
Note also the ++i instead of i++ which is a tiny tiny tiny bit faster